extensions/amp-analytics/0.1/events.js

```diff
@@ -389,6 +395,43 @@ export class VisibilityTracker extends EventTracker {
   }
 
   /**
+   * @param {string=} waitForSpec
+   * @param {string=} selector
+   * @param {Element=} element
+   * @return {?Promise}
+   * @visibleForTesting
+   */
+  getReadyPromise(waitForSpec, selector, element) {
+    if (waitForSpec == WAITFOR_EVENTS_LIST.NONE || !selector) {
+      // Always return null if specific asked waitFor NONE, or no selector
+      return null;
+    }
+    if (!element) {
+      // If selector is :root or :host, always wait for root ini-load
+      return this.iniLoadTracker_.getRootSignal();
+    }
+    if (!waitForSpec) {
+      // Default behavior for selector select AMP element is to wait for
+      // AMP element's ini-load
+      waitForSpec = WAITFOR_EVENTS_LIST.INI_LOAD;
+    }
+    if (typeof element.signals != 'function') {
+      return null;
+    }
+    switch (waitForSpec) {
+      case WAITFOR_EVENTS_LIST.INI_LOAD:
```

Comment:
An alternative idea: we have a map `<eventType, TrackerClass>` and we define an interface to resolve our signals. E.g. 
```
@interface
class SignalTrackerDef {
  getRootSignal(eventType): ?Promise;
  getElementSignal(eventType, element): ?Promise;
}

class SignalTracker implements SignalTrackerDef {...}
class IniLoadTracker implements SignalTrackerDef {...}
```

and then:
```
waitForTrackers = {
  'none': null,
  'render-start': SignalTracker,
  'ini-load': IniLoadTracker,
};
```

This way the correct details of finding a signal will be delegated to the actual tracker implementations and always guaranteed to be the same as events themselves.